File: week_1/day-2-prefix-sum-hashing/prefix/What_is_the_preprocessing_cost.js

function buildPrefix(arr) {
    const prefix = new Array(arr.length);
    prefix[0] = arr[0];

    for (let i = 1; i < arr.length; i++) {
        prefix[i] = prefix[i - 1] + arr[i];
    }

    return prefix;
}

const arr = [2, 4, 6, 8];
console.log(buildPrefix(arr));

File: week_1/day-2-prefix-sum-hashing/prefix/What_is_the_preprocessing_cost.py

arr = [2, 4, 6, 8]

def build_prefix(arr):
    prefix = [0] * len(arr)
    prefix[0] = arr[0]
    for i in range(1, len(arr)):
        prefix[i] = prefix[i - 1] + arr[i]
    return prefix

print(build_prefix(arr))
